mapcontrol.js

```diff
@@ -55,13 +55,97 @@ function createMapControl(elementName) {
         },
 
         addTask: function (coordinates, names) {
+            //Clearer if we don't show track to and from start line and finish line, as we are going to show lines
             var taskLayers = [L.polyline(coordinates, { color: 'blue' })];
+
+             var taskDrawOptions = {
+               color: 'green',
+               weight: 3,
+               opacity: 0.8
+             };
+             //definitions from BGA rules
+             //defined here as any future changes will be easier
+             var startLineRadius= 5;
+             var finishLineRadius= 1;
+             var tpCircleRadius= 500;
+             var tpSectorRadius= 20000;
+             var tpSectorAngle = 90;
             var j;
-            for (j = 0; j < coordinates.length; j++) {
+            for (j = 0; j  < coordinates.length; j++) {
                 taskLayers.push(L.marker(coordinates[j]).bindPopup(names[j]));
+                 switch(j)   {
+                    case 0:
+                         var startline=getline(coordinates[0],coordinates[1],startLineRadius,taskDrawOptions);
+                        taskLayers.push(startline);
+                        break;
+                    case (coordinates.length-1):
+                       var finishline= getline(coordinates[j], coordinates[j-1], finishLineRadius,taskDrawOptions);
+                        taskLayers.push(finishline);
+                        break;
+                    default:
+                        taskLayers.push(L.circle(coordinates[j], tpCircleRadius, taskDrawOptions));
+                         var tpsector=getTpSector(coordinates[j], coordinates[j-1],coordinates[j+1],tpSectorRadius,tpSectorAngle,taskDrawOptions);
+                        taskLayers.push(tpsector);
+                }
             }
             mapLayers.task = L.layerGroup(taskLayers).addTo(map);
             layersControl.addOverlay(mapLayers.task, 'Task');
         }
     };
+}    
+
+
+function getline(pt1,pt2,linerad,drawOptions) {
```

Comment:
Hi Alistair

Now fixed the bug and moved the functions (except "getBearing()), which 
I reckon is more generic.  See latest commit.

regards

On 02/09/15 21:31, Alistair Green wrote:
>
> In mapcontrol.js 
> <https://github.com/alistairmgreen/jsigc/pull/12#discussion_r38580413>:
>
> >              }
> >              mapLayers.task = L.layerGroup(taskLayers).addTo(map);
> >              layersControl.addOverlay(mapLayers.task, 'Task');
> >          }
> >      };
> > +}
> > +
> > +
> > +function getline(pt1,pt2,linerad,drawOptions) {
>
> I would be tempted to put these new functions inside 
> |createMapControl()| so that they don't have global scope.
>
> The `createMapControl
>
> —
> Reply to this email directly or view it on GitHub 
> <https://github.com/alistairmgreen/jsigc/pull/12/files#r38580413>.
>